rust/rsc/src/common/database.rs

```diff
@@ -190,9 +191,38 @@ pub async fn read_dbonly_blob_store<T: ConnectionTrait>(
 // ----------            Update            ----------
 
 // ----------            Delete            ----------
-pub async fn delete_all_jobs<T: ConnectionTrait>(db: &T) -> Result<DeleteResult, DbErr> {
+pub async fn delete_all_jobs<T: ConnectionTrait>(db: &T, chunk_size: u16) -> Result<u64, DbErr> {
     tracing::info!("Deleting ALL jobs");
-    job::Entity::delete_many().exec(db).await
+
+    if chunk_size == 0 {
+        panic!("Chunk size must be larger than zero");
+    }
+
+    let total = entity::job::Entity::find().count(db).await?;
+    let mut affected = 0;
+
+    loop {
+        let exec: ExecResult = db
+            .execute(Statement::from_sql_and_values(
+                DbBackend::Postgres,
+                r#"
+                DELETE FROM job
+                WHERE id IN
+                (
+                    SELECT id FROM job
+                    LIMIT $1
+                )
+                "#,
+                [chunk_size.into()],
+            ))
+            .await?;
+        affected += exec.rows_affected();
+        tracing::info!(%affected, %total, "Jobs deleted");
+
+        if exec.rows_affected() == 0 {
```

Comment:
Shouldn't this be `<= chunk_size`? Otherwise we do an extra delete when there are already no jobs.